services/api/src/managers/realmmanager.js

```diff
@@ -90,6 +94,12 @@ module.exports = {
       );
     }
 
+    // Hash all applications' clientSecret
+    for (const app of newRealm.applications) {
+      app.createdDate = new Date();
+      app.clientSecret = bcrypt.hashSync(app.clientSecret, 10);
+    }
+
     realmModel.add(newRealm, (errors, realm) => {
```

Comment:
This can be changed to use the mongoose model from `@microrealestate/common/models/realm`.

Also this code realmModel.schema.filter(req.body) can be removed as mongo will cleanup the payload before saving.